src/Client.php

```diff
@@ -92,49 +90,44 @@ class Client extends BaseClient
      *                                   string, e.g. "en-US" to fetch content in that locale. Set it to "*"
      *                                   to fetch content in all locales.
      * @param array       $options       An array of optional configuration options. The following keys are possible:
-     *                                   * guzzle      Override the guzzle instance used by the Contentful client
-     *                                   * logger      Inject a Contentful logger
-     *                                   * uriOverride Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
-     *                                   * cache       Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
-     *                                   * autoWarmup  Warm up the cache automatically
+     *                                   * guzzle     Override the guzzle instance used by the Contentful client
+     *                                   * logger     Inject a Contentful logger
+     *                                   * baseUri    Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
+     *                                   * cache      Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
+     *                                   * autoWarmup Warm up the cache automatically
      */
     public function __construct($token, $spaceId, $preview = false, $defaultLocale = null, array $options = [])
     {
-        $baseUri = $preview ? 'https://preview.contentful.com/' : 'https://cdn.contentful.com/';
-
         $options = \array_replace([
             'guzzle' => null,
             'logger' => null,
-            'uriOverride' => null,
+            'baseUri' => null,
```

Comment:
What's the reasoning for this change?

I agree that this is a better name, but this will add one more thing that users will have to have in mind if they want to upgrade to this new version.

Therefore, please create a guide for upgrading, and include this change in the `Changes` section of the changelog.

Reply:
There's already a `UPGRADE-3.0.md` file, I have a bunch of other stuff I have to add to it. It's not in this PR, but it's going to be there 🙂